test/infrastructure/docker/api/v1beta1/dockermachinetemplate_webhook.go

```diff
@@ -24,35 +24,44 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	clusteradmission "sigs.k8s.io/cluster-api/exp/admission"
 )
 
 const dockerMachineTemplateImmutableMsg = "DockerMachineTemplate spec.template.spec field is immutable. Please create a new resource instead."
 
 func (m *DockerMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	// register custom validator
+	mgr.GetWebhookServer().Register(
+		"/validate-infrastructure-cluster-x-k8s-io-v1beta1-dockermachinetemplate",
+		clusteradmission.TopologyAwareValidatingWebhookFor(m),
+	)
+	// register defaulter
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(m).
 		Complete()
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-dockermachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=dockermachinetemplates,versions=v1beta1,name=validation.dockermachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhook.Validator = &DockerMachineTemplate{}
+var _ clusteradmission.TopologyAwareValidator = &DockerMachineTemplate{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (m *DockerMachineTemplate) ValidateCreate() error {
 	return nil
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
-func (m *DockerMachineTemplate) ValidateUpdate(oldRaw runtime.Object) error {
+func (m *DockerMachineTemplate) ValidateUpdate(skipImmutability bool, oldRaw runtime.Object) error {
 	var allErrs field.ErrorList
 	old, ok := oldRaw.(*DockerMachineTemplate)
 	if !ok {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected a DockerMachineTemplate but got a %T", oldRaw))
 	}
-	if !reflect.DeepEqual(m.Spec.Template.Spec, old.Spec.Template.Spec) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec"), m, dockerMachineTemplateImmutableMsg))
+	if !skipImmutability {
+		if !reflect.DeepEqual(m.Spec.Template.Spec, old.Spec.Template.Spec) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec"), m, dockerMachineTemplateImmutableMsg))
+		}
 	}
```

Comment:
adjusted to a single line if, I'd rather not do the `if skipImmutability {return nil}` to prevent forgetting to validation which would be later on in the func.